packages/cli/src/config/config.ts

```diff
@@ -655,9 +662,24 @@ export async function loadCliConfig(
     codebaseInvestigatorSettings:
       settings.experimental?.codebaseInvestigatorSettings,
     fakeResponses: argv.fakeResponses,
+    recordResponses: argv.recordResponses,
     retryFetchErrors: settings.general?.retryFetchErrors ?? false,
     ptyInfo: ptyInfo?.name,
   });
+  // TODO: Consider moving `registerCleanup` into core, so this can move inside
+  // `createContentGenerator`.
+  const contentGenerator = config.getContentGenerator();
```

Comment:
I definitely don't love this part, suggestions welcome... streaming the output to the file constantly is I think not a great solution as the file format is not conducive to that. Although, the file format could be changed to be more friendly?